### DISCUSSION TITLE
Missed changes in watched directories (inotify?)

### DISCUSSION THREAD

**Discussion by aricooperdavis** [abraunegg/onedrive#2541] [General]
This morning I discovered that if I create files with `mogrify` within a directory watched by the onedrive client then those changes don't get "seen".

Can anyone else replicate this behaviour?

This may be a problem with inotify rather than this client, but I can't find any known-issues with inotify that may explain this behaviour.

```
$ onedrive --version
onedrive v2.4.25

$ uname -r
6.5.11-300.fc39.x86_64
```

**Comment by abraunegg** [comment]
@aricooperdavis 
> Can anyone else replicate this behaviour?

Please can you read https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support on how you can investigate this.

The application will detect changes on disk with `inotify` and when you are running in debug mode, you can see exactly what is being triggered.

My suggestion is that you investigate this first.

The second aspect is this - *where* is your 'sync_dir' pointing to? If your 'sync_dir' is pointing to a network mount point, then `inotify` *cannot* be used to detect the change as these mount points do not support 'inotify', thus tracking real-time changes via inotify of local files is not possible. As per the [documentation](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#sync_dir): 

![image](https://github.com/abraunegg/onedrive/assets/4956234/eeb275c2-acee-4036-8d62-3618f17178b7)


Outside of this, I would also suggest you look at testing 'alpha-3' as this has been 100% totally re-written, so if there is a 'watch' bug (unlikely, but you never know) potentially this version then has that fixed. Refer to https://github.com/abraunegg/onedrive/discussions/2521 for details. This does *not* change any 'inotify' lack of capability by network mount points however.

In either case if your 'sync_dir' is not a network mount point, please do some debugging. If `mogrify` does not write to the disk correctly to trigger an event, the issue is with `mogrify`.
